### PR TITLE
Vega title typings

### PR DESCRIPTION
**Pull request by domoritz**
Depends on #3993

### CODE REVIEW

File: build/vega-lite-schema.json [hunk 1]
```diff
@@ -1265,6 +1257,58 @@
       },
       "type": "object"
     },
+    "BaseTitle<number,string,Color,FontWeight,TextBaseline,TitleFrame,TitleAnchor>": {
```
Comment (kanitw) [inline build/vega-lite-schema.json]: Let's rename this. I'm sure Altair won't like this crazy generic.

File: site/docs/view/title.md [hunk 2]
```diff
@@ -38,6 +35,7 @@ For example, we can customize the `anchor` of the title of a bar chart.
 }
 ```
 
-To provide themes for all titles, the title config `config: {title: {...}}` can contain the following properties:
+{:#config}
+## Title Config
 
-{% include table.html props="anchor,angle,baseline,color,font,fontSize,fontWeight,limit,offset,orient" source="VgTitleConfig" %}
+To provide themes for all titles, the title configuration `config: {title: {...}}` supports all [title parameters](#params).
```
Comment (kanitw) [inline site/docs/view/title.md]: add parentheses to wrap the inline code block?

**Comment by kanitw** [comment]
Hmm test still failing.

File: src/title.ts [hunk 3]
```diff
@@ -28,7 +23,27 @@ export interface TitleBase {
    */
   style?: string | string[];
 
-  // TODO: name, encode, interactive, zindex
+  /**
+   * A boolean flag indicating if the title element should respond to input events such as mouse hover.
+   */
+  interactive?: boolean;
```
Comment (kanitw) [inline src/title.ts]: I don't think we expose `interactive` elsewhere in the codebase?

**Comment by domoritz** [comment]
Rebased and fixed tests.

File: src/title.ts (hunk 3 above)
Comment (domoritz) [reply src/title.ts]: Done

**Pull request merged**